receiver/otlpreceiver/otlp.go

```diff
@@ -110,7 +111,10 @@ func (r *otlpReceiver) startGRPCServer(host component.Host) error {
 		defer r.shutdownWG.Done()
 
 		if errGrpc := r.serverGRPC.Serve(gln); errGrpc != nil && !errors.Is(errGrpc, grpc.ErrServerStopped) {
-			r.settings.ReportStatus(component.NewFatalErrorEvent(errGrpc))
+			statusReporter, ok := host.(componentstatus.StatusReporter)
+			if ok {
+				statusReporter.ReportStatus(componentstatus.NewFatalErrorEvent(errGrpc))
+			}
```

Comment:
We could get some sort of wrapper like 

```go
func Report(host component.Host, e *componentstatus.StatusEvent) {
  statusReporter, ok := host.(componentstatus.StatusReporter)
  if ok {
    statusReporter.ReportStatus(e)
  }
}
```

to make this a bit more user-friendly (I think this was one of Matt's concerns)

Reply:
Great idea, added.